core/modules/replica/tools/qserv-replica-worker.cc

```diff
@@ -108,17 +120,55 @@ int main(int argc, const char* const argv[]) {
             argv,
             "\n"
             "Usage:\n"
-            "  <worker> [--config=<url>]\n"
+            "  <worker> [--db-allow-reconnect]\n"
+            "           [--db-reconnect-timeout=<sec>]\n"
+            "           [--db-max-retries=<num>]\n"
+            "           [--db-timeout=<sec>]\n"
+            "           [--config=<url>]\n"
             "\n"
             "Parameters:\n"
             "  <worker>   - the name of a worker\n"
             "\n"
             "Flags and options:\n"
-            "  --config   - a configuration URL (a configuration file or a set of the database\n"
-            "               connection parameters [ DEFAULT: file:replication.cfg ]\n");
+            "\n"
+            "    --db-allow-reconnect \n"
+            "\n"
+            "      change the default database connecton handling node. Set 0 to disable automatic\n"
+            "      reconnects. Any other number would man an opposite scenario.\n"
+            "      DEFAULT: " + std::to_string(::databaseAllowReconnect ? 1 : 0) + "\n"
+            "\n"
+            "    --db-reconnect-timeout \n"
+            "\n"
+            "      change the default value limiting a duration of time for making automatic\n"
+            "      reconnects to a database server before failing and reporting error (if the server\n"
+            "      is not up, or if it's not reachable for some reason)\n"
+            "      DEFAULT: " + std::to_string(::databaseConnectTimeoutSec) + "\n"
+            "\n"
+            "    --db-max-reconnects\n"
+            "\n"
+            "      change the default value limiting a number of attempts to repeat a sequence\n"
+            "      of queries due to connection losses and subsequent reconnects before to fail.\n"
+            "      DEFAULT: " + std::to_string(::databaseMaxReconnects) + "\n"
+            "\n"
+            "    --db-transaction-timeout \n"
+            "\n"
+            "      change the default value limiting a duration of each attempt to execute\n"
+            "      a database transaction before to fail.\n"
+            "      DEFAULT: " + std::to_string(::databaseTransactionTimeoutSec) + "\n"
+            "\n"
+            "    --config \n"
+            "\n"
+            "      onfiguration URL (a configuration file or a set of the database\n"
+            "      connection parameters [ DEFAULT: " + ::configUrl + " ]\n");
+
+        ::workerName = parser.parameter<std::string>(1);
+
+        ::databaseAllowReconnect        = parser.option<unsigned int>("db-allow-reconnect",     ::databaseAllowReconnect);
+        ::databaseConnectTimeoutSec     = parser.option<unsigned int>("db-reconnect-timeout",   ::databaseConnectTimeoutSec);
+        ::databaseMaxReconnects         = parser.option<unsigned int>("db-max-reconnects",      ::databaseMaxReconnects);
+        ::databaseTransactionTimeoutSec = parser.option<unsigned int>("db-transaction-timeout", ::databaseTransactionTimeoutSec);
```

Comment:
This looks like repeating pattern (including code above which declares these variables and updates config options, and that long command line string which repeats in every file), maybe worth factoring it out?

Reply:
Yes, I was already itching to do what you say. I will investigate options.

Reply:
Well, perhaps the best option here would be to introduce a base class for command-line application. The class would take care of mandate tasks, such as the one above, as well as a bunch of others. I'm not sure if I'm ready to introduce this class in a context of tis PR. Most probably, I'll do this in a separate ticket.